packages/headless/src/api/search/search-api-client.ts

```diff
@@ -99,14 +100,26 @@ export class SearchAPIClient {
     };
   }
 
+  private searchAbortController: AbortController | null = null;
+
   async search(
     req: SearchRequest
   ): Promise<SearchAPIClientResponse<SearchResponseSuccess>> {
+    if (this.searchAbortController) {
+      this.options.logger.warn('Cancelling current pending search query');
+      this.searchAbortController.abort();
```

Comment:
Hmm, so with this approach, we have a queue mechanism that will always use the most recent request. This is good. However, if requests are still being sent, then won't QPMs still be consumed?

Reply:
Yes, this part is not to prevent queries from being executed, but to abort previous queries and execute the most recent one. Like the JSUI does when "spamming" executeSearch()

Next PR(s) will prevent multiple successive fetchMoreResults on the result list, which is. a particularly dangerous use-case.